cache/infinispan/src/main/java/io/quarkus/ts/cache/infinispan/BaseServiceWithCache.java

```diff
@@ -0,0 +1,50 @@
+package io.quarkus.ts.cache.infinispan;
+
+import org.infinispan.protostream.GeneratedSchema;
+import org.infinispan.protostream.annotations.Proto;
+import org.infinispan.protostream.annotations.ProtoSchema;
+
+import io.quarkus.cache.CacheInvalidate;
+import io.quarkus.cache.CacheInvalidateAll;
+import io.quarkus.cache.CacheKey;
+import io.quarkus.cache.CacheResult;
+
+public abstract class BaseServiceWithCache {
+
+    private static final String CACHE_NAME = "service-cache";
+
+    private static int counter = 0;
```

Comment:
It seems, that this value may potentially be changed from the several threads. Please, change this variable to Atomic or another race-protected alternative